### ISSUE TITLE
options pair in useredit selectfield

### ISSUE THREAD

**Issue by JHeimbach** [ultimatemember/ultimatemember#115]
Hey,

im currently working on a selectfield which allows users to select another user as a value.

Preferably I will only save the id from the user (because the output is a mixture from more than one field) and build the output trough a filter (that works so far).

In the editsection, the array what i build (trough a filter) won't work, i figured out why, my only hope is that $options_pair is anywhere specified so that $k from my array will be the value.

TL;DR
is there somewhere the possibility that i can key,value pair use as a select option?

If there is no solution currently, than i have two questions:
1. Why is there [these](https://github.com/ultimatemember/ultimatemember/blob/master/core/um-fields.php#L1568-L1570) lines in the code?
2. Can someone push me in the right direction how i can the key,value pair make working?

Greets,
JHeimbach

**Comment by champsupertramp** [comment]
Hi,

Did you try this filter hook? 
https://github.com/ultimatemember/ultimatemember/blob/master/core/um-fields.php#L936

**Comment by JHeimbach** [comment]
Hey,

after you mentioned it, i have tried it but this is no solution, I only get the type in this filter, how could that help me?

**Comment by champsupertramp** [comment]
Hi,

You can declare a custom type and break the 'select' field type: 
https://github.com/ultimatemember/ultimatemember/blob/master/core/um-fields.php#L941

```PHP 
add_filter('um_hook_for_field_select','um_hook_break_field_select');
function um_hook_break_field_select( $type ){
     $type = 'my_custom_type'; //  
    return $type;
} 
```
Once you've changed the select type to your custom type, This https://github.com/ultimatemember/ultimatemember/blob/master/core/um-fields.php#L944 will use the default.

Now, you can catch the output with this filter: https://github.com/ultimatemember/ultimatemember/blob/master/core/um-fields.php#L949

```PHP
add_filter('um_edit_field_edit_my_custom_type','um_edit_field_edit_my_custom_type', 2,10);
function um_edit_field_edit_my_custom_type( $output, $data){
     if( isset( $data['key'] ) &&   $data['key']  == 'my_custom_select_meta_key'){
           $output = 'Do something'; //
     } 
   return $output;
} 

```

**Comment by shramee** [comment]
dfhg
